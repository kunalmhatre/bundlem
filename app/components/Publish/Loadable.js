/**
 *
 * Asynchronously loads the component for Publish
 *
 */

import loadable from 'loadable-components';

export default loadable(() => import('./index'));
