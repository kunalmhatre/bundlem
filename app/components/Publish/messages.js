/*
 * Publish Messages
 *
 * This contains all the text for the Publish component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Publish';

export default defineMessages({
  publishTitle: {
    id: `${scope}.publishTitle`,
    defaultMessage: 'Congrats, your bundle has been published!',
  },
  bundleID: {
    id: `${scope}.bundleID`,
    defaultMessage: 'BUNDLE ID',
  },
  viewButton: {
    id: `${scope}.viewButton`,
    defaultMessage: 'VIEW',
  },
  copyLinkButton: {
    id: `${scope}.copyLinkButton`,
    defaultMessage: 'COPY LINK',
  },
  thankYouMessage: {
    id: `${scope}.thankYouMessage`,
    defaultMessage: 'Thank you for using Bundlem',
  },
  copiedTooltip: {
    id: `${scope}.copiedTooltip`,
    defaultMessage: 'Copied',
  },
});
