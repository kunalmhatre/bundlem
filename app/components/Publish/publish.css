.publish-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    font-weight: 700;
    flex-grow: 1;
    justify-content: center;
}

.publish-title {
    font-size: 2rem;
    padding: 1rem;
    text-align: center;
}

.publish-bundle-id-title {
    text-transform: uppercase;
    padding: 0.5rem;
    font-size: 1.5rem;
    letter-spacing: 0.1rem;
}

.publish-bundle-id {
    font-size: 5rem;
}

.publish-button-group {
    display: flex;
    flex-direction: row;
    width: 18rem;
    justify-content: space-around;
    padding: 0.5rem;
}

.publish-thank-you-message {
    font-size: 1.5rem;
    margin: 3rem 0 1.5rem 0;
    padding: 2rem 7rem 2rem 7rem;
    border: 2px solid;
}

@media only screen and (max-width: 767px) {
    .publish-title {
        font-size: 1.3rem;
        margin-top: 3rem;
    }
    .publish-bundle-id-title {
        font-size: 1rem;
    }
    .publish-bundle-id {
        font-size: 4rem;
    }
    .publish-button-group {
        width: 14rem;
    }
    .publish-thank-you-message {
        font-size: 0.9rem;
        margin-top: 1.5rem;
        padding: 1rem 3.5rem 1rem 3.5rem;
    }
}

@media only screen and (min-width: 768px) and (max-width: 991px) {
    .publish-title {
        margin-top: 3rem;
    }
}

@media only screen and (max-width: 360px) {
    .publish-title {
        font-size: 1.2rem;
        margin-top: 1.5rem;
    }
    .publish-button-group {
        width: 12rem;
    }
    .publish-thank-you-message {
        padding: 1rem 2rem 1rem 2rem;
    }
}