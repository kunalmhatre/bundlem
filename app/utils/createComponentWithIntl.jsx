/**
 * Renders component with react-intl support for snapshot testing
 * Source: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#helper-function-2
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

const createComponentWithIntl = (children, props = { locale: 'en' }) => renderer.create(
  <IntlProvider {...props}>
    {children}
  </IntlProvider>,
);

export default createComponentWithIntl;
