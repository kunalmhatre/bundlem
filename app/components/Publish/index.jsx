import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Tooltip } from 'antd';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import Button from '../Button';
import PageLayout from '../PageLayout';
import messages from './messages';
import './publish.css';

const propTypes = {
  bundleID: PropTypes.number,
};

const defaultProps = {
  bundleID: 0,
};

/**
 * Publish component
 * @param {number} bundleID - ID received for the bundle after publishing
 */
function Publish({ bundleID }) {
  return (
    <PageLayout>
      <div className="publish-content">
        <div className="publish-title">
          <FormattedMessage {...messages.publishTitle} />
        </div>
        <div className="publish-bundle-id-title">
          <FormattedMessage {...messages.bundleID} />
        </div>
        <div className="publish-bundle-id">
          { bundleID }
        </div>
        <div className="publish-button-group">
          <span><Button text={messages.viewButton} /></span>
          <CopyToClipboard text={`https://bundlem.in/bundle/${bundleID}`}>
            <Tooltip
              title={<FormattedMessage {...messages.copiedTooltip} />}
              trigger="click"
            >
              <span><Button text={messages.copyLinkButton} /></span>
            </Tooltip>
          </CopyToClipboard>
        </div>
        <div className="publish-thank-you-message">
          <FormattedMessage {...messages.thankYouMessage} />
        </div>
      </div>
    </PageLayout>
  );
}

Publish.propTypes = propTypes;
Publish.defaultProps = defaultProps;

export default Publish;
