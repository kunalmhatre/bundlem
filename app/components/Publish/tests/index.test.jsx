import React from 'react';

import createComponentWithIntl from '../../../../internals/testing/createComponentWithIntl';
import Publish from '../index';

describe('<Publish />', () => {
  it('renders old snapshot', () => {
    expect(
      createComponentWithIntl(
        <Publish />,
      ).toJSON(),
    ).toMatchSnapshot();
  });
});
