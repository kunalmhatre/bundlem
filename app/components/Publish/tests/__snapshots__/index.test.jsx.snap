// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Publish /> renders old snapshot 1`] = `
<div
  className="ant-row page-layout theme-blue"
>
  <div
    className="ant-col-22 ant-col-offset-1"
  >
    <div
      className="page-layout-header-content"
    >
      <header
        className="header theme-blue"
      >
        <span>
          Bundlem
        </span>
      </header>
      <section
        className="page-layout-content"
      >
        <div
          className="publish-content"
        >
          <div
            className="publish-title"
          >
            <span>
              Congrats, your bundle has been published!
            </span>
          </div>
          <div
            className="publish-bundle-id-title"
          >
            <span>
              BUNDLE ID
            </span>
          </div>
          <div
            className="publish-bundle-id"
          >
            0
          </div>
          <div
            className="publish-button-group"
          >
            <span>
              <button
                className="button theme-blue-inverse theme-blue-border"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span>
                  VIEW
                </span>
                
              </button>
            </span>
            <span
              onClick={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <button
                className="button theme-blue-inverse theme-blue-border"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span>
                  COPY LINK
                </span>
                
              </button>
            </span>
          </div>
          <div
            className="publish-thank-you-message"
          >
            <span>
              Thank you for using Bundlem
            </span>
          </div>
        </div>
      </section>
    </div>
    <footer
      className="footer theme-blue"
    >
      <a
        className="footer-link theme-blue"
        href="https://github.com/kunalmhatre/bundlem/"
      >
        <span>
          source
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://github.com/kunalmhatre/bundlem/issues/new/choose"
      >
        <span>
          REPORT ISSUE
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://kunalmhatre.com/bundlem/"
      >
        <span>
          STORY
        </span>
      </a>
      <a
        className="footer-link theme-blue"
        href="https://kunalmhatre.com/"
      >
        <span>
          AUTHOR
        </span>
      </a>
    </footer>
  </div>
</div>
`;
